### ISSUE TITLE
java.io.NotSerializableException : gwtupload.server.UploadListener

### ISSUE THREAD

**Issue by GoogleCodeExporter** [bdshadow/gwtupload#7]
```
What steps will reproduce the problem?
1.Select a file to upload
2.Press the send Button
3.Nothing happens, check logs on Google App Server

What is the expected output? What do you see instead?
file to uploaded and processed. No issue when run locally. Log when 
deployed to google app shows
#

220.227.77.169 - admin_mail_reception [03/Sep/2009:09:30:10 -0700] "POST
/mis_snail_mail_tracker/fileupload HTTP/1.1" 500 4624
"http://smailtracker.appspot.com/" "Mozilla/5.0 (Windows; U; Windows NT
6.0; en-US; rv:1.9.1.2) Gecko/20090729 Firefox/3.5.2,gzip(gfe)"
"smailtracker.appspot.com"

#
W 09-03 09:30AM 10.343

/mis_snail_mail_tracker/fileupload
java.lang.RuntimeException: java.io.NotSerializableException:
gwtupload.server.UploadListener
    at
com.google.apphosting.runtime.jetty.SessionManager.serialize(SessionManager.java
:334)
    at
com.google.apphosting.runtime.jetty.SessionManager.createEntityForSession(Sessio
nManager.java:314)
    at
com.google.apphosting.runtime.jetty.SessionManager$AppEngineSession.save(Session
Manager.java:153)
    at
com.google.apphosting.runtime.jetty.SaveSessionFilter.doFilter(SaveSessionFilter
.java:41)
    at
org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.jav
a:1084)
    at
com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(Transactio
nCleanupFilter.java:43)
    at
org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.jav
a:1084)
    at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:360)
    at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
    at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:181)
    at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:712)
    at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:405)
    at
com.google.apphosting.runtime.jetty.AppVersionHandlerMap.handle(AppVersionHandle
rMap.java:237)
    at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:139)
    at org.mortbay.jetty.Server.handle(Server.java:313)
    at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:506)
    at
org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.ja
va:830)
    at
com.google.apphosting.runtime.jetty.RpcRequestParser.parseAvailable(RpcRequestPa
rser.java:76)
    at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:381)
    at
com.google.apphosting.runtime.jetty.JettyServletEngineAdapter.serviceRequest(Jet
tyServletEngineAdapter.java:139)
    at
com.google.apphosting.runtime.JavaRuntime.handleRequest(JavaRuntime.java:235)
    at
com.google.apphosting.base.RuntimePb$EvaluationRuntime$6.handleBlockingRequest(R
untimePb.java:4823)
    at
com.google.apphosting.base.RuntimePb$EvaluationRuntime$6.handleBlockingRequest(R
untimePb.java:4821)
    at
com.google.net.rpc.impl.BlockingApplicationHandler.handleRequest(BlockingApplica
tionHandler.java:24)
    at com.google.net.rpc.impl.RpcUtil.runRpcInApplication(RpcUtil.java:359)
    at com.google.net.rpc.impl.Server$2.run(Server.java:820)
    at
com.google.tracing.LocalTraceSpanRunnable.run(LocalTraceSpanRunnable.java:56)
    at
com.google.tracing.LocalTraceSpanBuilder.internalContinueSpan(LocalTraceSpanBuil
der.java:516)
    at com.google.net.rpc.impl.Server.startRpc(Server.java:775)
    at com.google.net.rpc.impl.Server.processRequest(Server.java:348)
    at
com.google.net.rpc.impl.ServerConnection.messageReceived(ServerConnection.java:4
36)
    at com.google.net.rpc.impl.RpcConnection.parseMessages(RpcConnection.java:319)
    at com.google.net.rpc.impl.RpcConnection.dataReceived(RpcConnection.java:290)
    at com.google.net.async.Connection.handleReadEvent(Connection.java:428)
    at
com.google.net.async.EventDispatcher.processNetworkEvents(EventDispatcher.java:7
62)
    at com.google.net.async.EventDispatcher.internalLoop(EventDispatcher.java:207)
    at com.google.net.async.EventDispatcher.loop(EventDispatcher.java:101)
    at com.google.net.rpc.RpcService.runUntilServerShutdown(RpcService.java:251)
    at
com.google.apphosting.runtime.JavaRuntime$RpcRunnable.run(JavaRuntime.java:374)
    at java.lang.Thread.run(Unknown Source)
Caused by: java.io.NotSerializableException: gwtupload.server.UploadListener
    at java.io.ObjectOutputStream.writeObject0(Unknown Source)
    at java.io.ObjectOutputStream.writeObject(Unknown Source)
    at java.util.HashMap.writeObject(Unknown Source)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at java.io.ObjectStreamClass.invokeWriteObject(Unknown Source)
    at java.io.ObjectOutputStream.writeSerialData(Unknown Source)
    at java.io.ObjectOutputStream.writeOrdinaryObject(Unknown Source)
    at java.io.ObjectOutputStream.writeObject0(Unknown Source)
    at java.io.ObjectOutputStream.writeObject(Unknown Source)
    at
com.google.apphosting.runtime.jetty.SessionManager.serialize(SessionManager.java
:331)
    ... 39 more

#
C 09-03 09:30AM 10.346

Uncaught exception from servlet
java.lang.RuntimeException: java.io.NotSerializableException:
gwtupload.server.UploadListener
    at
com.google.apphosting.runtime.jetty.SessionManager.serialize(SessionManager.java
:334)
    at
com.google.apphosting.runtime.jetty.SessionManager.createEntityForSession(Sessio
nManager.java:314)
    at
com.google.apphosting.runtime.jetty.SessionManager$AppEngineSession.save(Session
Manager.java:153)
    at
com.google.apphosting.runtime.jetty.SaveSessionFilter.doFilter(SaveSessionFilter
.java:41)
    at
org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.jav
a:1084)
    at
com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(Transactio
nCleanupFilter.java:43)
    at
org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.jav
a:1084)
    at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:360)
    at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
    at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:181)
    at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:712)
    at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:405)
    at
com.google.apphosting.runtime.jetty.AppVersionHandlerMap.handle(AppVersionHandle
rMap.java:237)
    at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:139)
    at org.mortbay.jetty.Server.handle(Server.java:313)
    at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:506)
    at
org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.ja
va:830)
    at
com.google.apphosting.runtime.jetty.RpcRequestParser.parseAvailable(RpcRequestPa
rser.java:76)
    at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:381)
    at
com.google.apphosting.runtime.jetty.JettyServletEngineAdapter.serviceRequest(Jet
tyServletEngineAdapter.java:139)
    at
com.google.apphosting.runtime.JavaRuntime.handleRequest(JavaRuntime.java:235)
    at
com.google.apphosting.base.RuntimePb$EvaluationRuntime$6.handleBlockingRequest(R
untimePb.java:4823)
    at
com.google.apphosting.base.RuntimePb$EvaluationRuntime$6.handleBlockingRequest(R
untimePb.java:4821)
    at
com.google.net.rpc.impl.BlockingApplicationHandler.handleRequest(BlockingApplica
tionHandler.java:24)
    at com.google.net.rpc.impl.RpcUtil.runRpcInApplication(RpcUtil.java:359)
    at com.google.net.rpc.impl.Server$2.run(Server.java:820)
    at
com.google.tracing.LocalTraceSpanRunnable.run(LocalTraceSpanRunnable.java:56)
    at
com.google.tracing.LocalTraceSpanBuilder.internalContinueSpan(LocalTraceSpanBuil
der.java:516)
    at com.google.net.rpc.impl.Server.startRpc(Server.java:775)
    at com.google.net.rpc.impl.Server.processRequest(Server.java:348)
    at
com.google.net.rpc.impl.ServerConnection.messageReceived(ServerConnection.java:4
36)
    at com.google.net.rpc.impl.RpcConnection.parseMessages(RpcConnection.java:319)
    at com.google.net.rpc.impl.RpcConnection.dataReceived(RpcConnection.java:290)
    at com.google.net.async.Connection.handleReadEvent(Connection.java:428)
    at
com.google.net.async.EventDispatcher.processNetworkEvents(EventDispatcher.java:7
62)
    at com.google.net.async.EventDispatcher.internalLoop(EventDispatcher.java:207)
    at com.google.net.async.EventDispatcher.loop(EventDispatcher.java:101)
    at com.google.net.rpc.RpcService.runUntilServerShutdown(RpcService.java:251)
    at
com.google.apphosting.runtime.JavaRuntime$RpcRunnable.run(JavaRuntime.java:374)
    at java.lang.Thread.run(Unknown Source)
Caused by: java.io.NotSerializableException: gwtupload.server.UploadListener
    at java.io.ObjectOutputStream.writeObject0(Unknown Source)
    at java.io.ObjectOutputStream.writeObject(Unknown Source)
    at java.util.HashMap.writeObject(Unknown Source)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at java.io.ObjectStreamClass.invokeWriteObject(Unknown Source)
    at java.io.ObjectOutputStream.writeSerialData(Unknown Source)
    at java.io.ObjectOutputStream.writeOrdinaryObject(Unknown Source)
    at java.io.ObjectOutputStream.writeObject0(Unknown Source)
    at java.io.ObjectOutputStream.writeObject(Unknown Source)
    at
com.google.apphosting.runtime.jetty.SessionManager.serialize(SessionManager.java
:331)
    ... 39 more



What version of the product are you using? On what operating system?
latest 0.3 on vista

Please provide any additional information below.
No issue when run locally. Problem happened when deployed and run from
Google Apps

```

Original issue reported on code.google.com by `kbin...@gmail.com` on 3 Sep 2009 at 4:53

**Comment by GoogleCodeExporter** [comment]
```
So far gwtupload doesnt work in app-engine because security issues in writing 
to file
system. So the file has to be saved in memory or in datastore.

There is already in the trunk a servlet for google application engine in which 
I'm
working but unfortunately it doesn't work yet.


Thanks
Manolo
```

Original comment by `manuel.carrasco.m` on 13 Sep 2009 at 10:10
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Hello, 

I've released version 0.5.3, it supports be deployed in AppEngine. 
Although the new release includes new stuff specific for AppEngine, 
UploadListener is
serializable and the reported error doesn't happen. 

Please take a look to the new version

Thank you very much for reporting

Manolo Carrasco
```

Original comment by `manuel.carrasco.m` on 11 Oct 2009 at 8:43
- Changed state: **Fixed**
